bin/check_security

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+print_usage() {
+    echo "Security Scanner"
+    echo " "
+    echo "Description:"
+    echo "Runs gosec on directories which git detects and marks in the diff."
+    echo "If the branch is detected as 'master', it will scan all"
+    echo "directories regardless of what has been modified locally."
+    echo " "
+    echo "Format:"
+    echo "security_scan [arguments]"
+    echo " "
+    echo "Options:"
+    echo "-h, --help                Show help"
+    echo "-c                        Specify the minimum confidence gosec needs to report an issue."
+    echo "-s                        Specify the minimum severity gosec needs to report an issue"
+    exit 0
+}
+
+# Default values for gosec
+confidence='medium'
+severity='high'
+current_branch=''
+
+while getopts 'b:c:s:h' flag; do
+  case "${flag}" in
+    b) current_branch="${OPTARG}" ;;
+    c) confidence="${OPTARG}" ;;
+    s) severity="${OPTARG}" ;;
+    h) print_usage ;;
+    *) print_usage ;;
+  esac
+done
+
+# This script is designed to be ran within the Secretless-Broker repository,
+# to use elsewhere, you will need to change this.
+working_directory=${PWD}/..
```

Comment:
You should use the abspath [here](https://github.com/cyberark/secretless-broker/blob/master/bin/abspath) to figure out the path. Generally you also want to use the program `pwd` rather than the variable `PWD` if you want to use pwd.

Reply:
If you leave as-is, you need some quotes